plugins/default/src/Ide/Plugin/Eval/CodeLens.hs

```diff
@@ -0,0 +1,694 @@
+                                                 modifyFlags)
+import           Ide.Plugin.Eval.Parse.Option   (langOptions)
+import           Ide.Plugin.Eval.Parse.Section  (Section (sectionFormat, sectionTests),
+                                                 allSections)
+import           Ide.Plugin.Eval.Parse.Token    (tokensFrom)
+import           Ide.Plugin.Eval.Types          (Format (SingleLine), Loc,
+                                                 Located (Located), Test,
+                                                 hasTests, isProperty,
+                                                 splitSections, unLoc)
+import           Ide.Types                      (CodeLensProvider,
+                                                 CommandFunction, CommandId,
+                                                 PluginCommand (PluginCommand))
+import           Language.Haskell.LSP.Core      (LspFuncs (getVirtualFileFunc, withIndefiniteProgress),
+                                                 ProgressCancellable (Cancellable))
+import           Language.Haskell.LSP.Types     (ApplyWorkspaceEditParams (ApplyWorkspaceEditParams),
+                                                 CodeLens (CodeLens),
+                                                 CodeLensParams (CodeLensParams, _textDocument),
+                                                 Command (_arguments, _title),
+                                                 ErrorCode (InternalError),
+                                                 List (List), Range (Range),
+                                                 ResponseError (ResponseError),
+                                                 ServerMethod (WorkspaceApplyEdit),
+                                                 TextDocumentIdentifier (..),
+                                                 TextEdit (TextEdit),
+                                                 WorkspaceEdit (WorkspaceEdit),
+                                                 toNormalizedUri)
+import           Language.Haskell.LSP.VFS       (virtualFileText)
+import           Outputable                     (nest, ppr, showSDoc, text,
+                                                 ($$), (<+>))
+import           System.FilePath                (takeFileName)
+import           System.IO                      (hClose)
+import           System.IO.Temp                 (withSystemTempFile)
+import           Text.Read                      (readMaybe)
+import           Util                           (OverridingBool (Never))
+
+-- | Code Lens provider
+-- NOTE: Invoked every time the document is modified, not just when the document is saved.
+codeLens :: CodeLensProvider
+codeLens lsp _state plId CodeLensParams {_textDocument} = response $ do
+  now <- liftIO getCurrentTime
+  let TextDocumentIdentifier uri = _textDocument
+  contents <- liftIO $ getVirtualFileFunc lsp $ toNormalizedUri uri
+  let Just text = virtualFileText <$> contents
```

Comment:
Please fix this incomplete pattern match

Reply:
fixed in latest commit